src/apps/invitations/crud.py

```diff
@@ -130,6 +130,48 @@ async def get_pending_by_invitor_id(self, user_id: uuid.UUID, query_params: Quer
             )
         return results
 
+    async def get_latest_by_emails(self, emails: list[str]) -> dict[str, InvitationDetail]:
```

Comment:
Just wondering how we are guaranteeing that the _latest_ invitation is the one that gets assigned to the `results` dictionary, without any SQL ordering happening on the invitation date? Are we just relying on other logic in the app that we're sure is always deleting old invitations?

Reply:
It's pretty much depending on other application logic. Users only have one pending/approved invitation per applet. Rejected invitations are set to a `declined` status, so practically this result set will contain the user's latest invitation for each applet